compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1777,9 +1780,13 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         // point we may encounter a candidate where the test is not relevant; at that point, we stop
         // sorting.
         while let Some(candidate) = candidates.first_mut() {
-            let Some(branch) =
-                self.sort_candidate(match_place, test, candidate, &target_candidates)
-            else {
+            let Some(branch) = self.sort_candidate(
+                match_place,
+                test,
+                candidate,
+                &target_candidates,
+                &mut catch_full_matched,
+            ) else {
                 break;
             };
             let (candidate, rest) = candidates.split_first_mut().unwrap();
```

Comment:
Below this, if `target_candidates.entry(branch)` is empty you know this is the first candidate for this branch, so if there was a span return you can add it to the new map.